mlflow/exceptions.py

```diff
@@ -22,11 +23,21 @@
     ErrorCode.Name(ENDPOINT_NOT_FOUND): 404,
     ErrorCode.Name(RESOURCE_DOES_NOT_EXIST): 404,
     ErrorCode.Name(PERMISSION_DENIED): 403,
+    ErrorCode.Name(CUSTOMER_UNAUTHORIZED): 401,
     ErrorCode.Name(BAD_REQUEST): 400,
     ErrorCode.Name(RESOURCE_ALREADY_EXISTS): 400,
     ErrorCode.Name(INVALID_PARAMETER_VALUE): 400,
 }
 
+HTTP_STATUS_TO_ERROR_CODE = dict((v, k) for k, v in ERROR_CODE_TO_HTTP_STATUS.items())
+HTTP_STATUS_TO_ERROR_CODE[404] = ErrorCode.Name(RESOURCE_DOES_NOT_EXIST)
```

Comment:
@balvisio Looks great! One more small ask: could we explicitly map `500` to `INTERNAL_ERROR` and `400` to `BAD_REQUEST`?

Reply:
@dbczumar Updated the PR.